### PR TITLE
updating structure run result to return BaseArtifact

### PR DESCRIPTION
**Pull request by emjay07**
- [X] I have read and agree to the contributing guidelines for [submitting new pull requests](https://github.com/griptape-ai/griptape?tab=readme-ov-file#submitting-pull-requests).

## Describe your changes
- Update output Artifact type from structure runs to `BaseArtifact`

## Issue ticket number and link
N/A

### CODE REVIEW

**Review by collindutter — COMMENTED** [review]
Thanks! Can you please update the changelog?

**Review by collindutter — APPROVED** [review]

File: griptape/drivers/structure_run/griptape_cloud_structure_run_driver.py [hunk 1]
```diff
@@ -44,7 +44,7 @@ def try_run(self, *args: BaseArtifact) -> BaseArtifact:
         except (exceptions.RequestException, HTTPError) as err:
             return ErrorArtifact(str(err))
 
-    def _get_structure_run_result(self, structure_run_id: str) -> InfoArtifact | TextArtifact | ErrorArtifact:
+    def _get_structure_run_result(self, structure_run_id: str) -> InfoArtifact | BaseArtifact | ErrorArtifact:
```
Comment (vachillo) [inline griptape/drivers/structure_run/griptape_cloud_structure_run_driver.py:47]: should `InfoArtifact` and `ErrorArtifact` be removed from the type hint?

Comment (emjay07) [reply griptape/drivers/structure_run/griptape_cloud_structure_run_driver.py:47]: Those are explicitly separate code paths, so I thought it was useful to leave them. I do know they inherit from BaseArtifact though, so *technically* they don't need to be there.

Comment (collindutter) [reply griptape/drivers/structure_run/griptape_cloud_structure_run_driver.py:47]: That's how I reasoned about it too, I think we should keep them.

**Pull request merged**